ml-proto/runtests.py

```diff
@@ -15,9 +15,14 @@ def auxFile(path):
     return path
 
 class RunTests(unittest.TestCase):
-  def _runCommand(self, command):
-    exitCode = subprocess.call(command, shell=True)
-    self.assertEqual(0, exitCode, "test runner failed with exit code %i" % exitCode)
+  def _runCommand(self, command, logPath = None, expectedExitCode = 0):
+    out = None if logPath is None else file(logPath, 'w+')
+
+    exitCode = subprocess.call(command, shell=True, stdout=out, stderr=subprocess.STDOUT)
+    self.assertEqual(expectedExitCode, exitCode, "test runner failed with exit code %i, expected %i" % (exitCode, expectedExitCode))
+
+    if logPath is not None:
+        out.close()
```

Comment:
One further style nit I just noted: inconsistent indentation, rest of the file uses 2